lib/spack/spack/solver/concretize.lp

```diff
@@ -7,31 +7,112 @@
 % This logic program implements Spack's concretizer
 %=============================================================================
 
+#const root_node_id = 0.
```

Comment:
I think this is confusing b/c it's overloading the word `root`.  Is there something else we could call this?  `start_node_id`?  `base_node_id`? `initial_replica_id`?  I can't think of anything particularly good but something other than root seems needed.

Reply:
Done in https://github.com/spack/spack/pull/38447/commits/89e7aa990b09df631ed755aeaf676a11f803ac62